inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/multiple/WholeDatabaseMigrationOperator.java

```diff
@@ -0,0 +1,228 @@
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.Transaction;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.SupportsNamespaces;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.flink.CatalogLoader;
+import org.apache.iceberg.flink.FlinkSchemaUtil;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.types.Types.NestedField;
+import org.apache.inlong.sort.base.format.AbstractDynamicSchemaFormat;
+import org.apache.inlong.sort.base.format.DynamicSchemaFormatFactory;
+import org.apache.inlong.sort.base.sink.MultipleSinkOption;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+
+public class WholeDatabaseMigrationOperator extends AbstractStreamOperator<RecordWithSchema>
+        implements OneInputStreamOperator<RowData, RecordWithSchema> {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    private final CatalogLoader catalogLoader;
+    private final MultipleSinkOption multipleSinkOption;
+
+    private transient Catalog catalog;
+    private transient SupportsNamespaces asNamespaceCatalog;
+    private transient AbstractDynamicSchemaFormat<JsonNode> dynamicSchemaFormat;
+
+    // record cache, wait schema to consume record
+    private transient Map<TableIdentifier, Queue<RecordWithSchema>> recordQueues;
+
+    // schema cache
+    private transient Map<TableIdentifier, Schema> schemaCache;
+
+    public WholeDatabaseMigrationOperator(CatalogLoader catalogLoader,
+            MultipleSinkOption multipleSinkOption) {
+        this.catalogLoader = catalogLoader;
+        this.multipleSinkOption = multipleSinkOption;
+    }
+
+    @Override
+    public void open() throws Exception {
+        super.open();
+        this.catalog = catalogLoader.loadCatalog();
+        this.asNamespaceCatalog =
+                catalog instanceof SupportsNamespaces ? (SupportsNamespaces) catalog : null;
+        this.recordQueues = new HashMap<>();
+        this.schemaCache = new HashMap<>();
+        this.dynamicSchemaFormat = DynamicSchemaFormatFactory.getFormat(multipleSinkOption.getFormat());
+    }
+
+    @Override
+    public void close() throws Exception {
+        super.close();
+        if (catalog instanceof Closeable) {
+            ((Closeable) catalog).close();
+        }
+    }
+
+    @Override
+    public void processElement(StreamRecord<RowData> element) throws Exception {
+        String wholeData = element.getValue().getString(0).toString();

```

Comment:
May be it can get the data by 'dynamicSchemaFormat.deserialize(rowData.getBinary(0))'?

Reply:
change it